### PR TITLE
Run setup commands inline rather than writing them to a script and running the script.

### LINKED ISSUE #1398: Use exclusive write lock when creating script files
To avoid changes to the file before it has executed.

### PR DESCRIPTION
**Pull request by ericpromislow**
Less code, and no need to create a temporary script.

Fixes #1398

### CODE REVIEW

File: src/k8s-engine/lima.ts [hunk 1]
```diff
@@ -726,33 +726,19 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
       return;
     }
     await this.showSudoReason(explanations);
+    const singleCommand = commands.join('; ');
```
Comment (mook-as) [inline src/k8s-engine/lima.ts:729]: We can still have `exec &> >(tee ${ logfile })` & `set -ex` in the command list (to get the logging), right?

File: src/k8s-engine/lima.ts [hunk 2]
```diff
@@ -726,33 +726,19 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
...
-
-${ commands.join('\n') }
-`,
-    { mode: 0o700 });
+    if (singleCommand.includes("'")) {
+      throw new Error(`Can't execute commands ${ singleCommand } because there's a single-quote in them.`);
+    }
     try {
-      await this.sudoExec(tmpScript);
+      await this.sudoExec(`/bin/sh -c '${ singleCommand }'`);
```
Comment (mook-as) [inline src/k8s-engine/lima.ts:735]: We're switching from `/usr/bin/env bash` to `/bin/sh` here; is that okay?

File: src/k8s-engine/lima.ts [hunk 3]
```diff
@@ -726,33 +726,19 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
...
-    await fs.promises.writeFile(tmpScript, `#!/usr/bin/env bash
-
-exec &> >(tee ${ logFile })
-set -ex
-
-${ commands.join('\n') }
-`,
-    { mode: 0o700 });
+    if (singleCommand.includes("'")) {
+      throw new Error(`Can't execute commands ${ singleCommand } because there's a single-quote in them.`);
```
Comment (mook-as) [inline src/k8s-engine/lima.ts:732]: 👍 I wrote a comment about adding this check… and saw that it's already there :D

**Review by mook-as — CHANGES_REQUESTED** [review]

File: src/k8s-engine/lima.ts (hunk 2 above)
Comment (jandubois) [reply src/k8s-engine/lima.ts:735]: I think so. Anything you run via `sudo` should not rely on the environment to look up executables.

The commands should all be basic POSIX shell commands. At least in our Alpine image `bash` is just a wrapper around `/bin/sh`:

```
$ rdlima cat /bin/bash
#!/bin/sh
exec /bin/ash "$@"
```

File: src/k8s-engine/lima.ts (hunk 1 above)
Comment (ericpromislow) [reply src/k8s-engine/lima.ts:729]: I'm not sure about it, which is why I left it out.

Running `bash -c 'CMD1;CMD2;CMD3…'`

is more like sourcing a file containing those commands than exec'ing them (e.g. `./run-script.sh`), and I don't like messing with the current session like that.

Although the "current session" is just a single `ssh` session to run the commands, so maybe... I'll have a look.

Comment (ericpromislow) [reply src/k8s-engine/lima.ts:729]: Just added `ex` to the `/bin/sh` command -- no need to manage the output in a separate log file when the output gets echoed to `lima.log`

**Review by jandubois — COMMENTED** [review]
LGTM, but I haven't actually tested it. Lmk if I should!

**Review by mook-as — APPROVED** [review]

**Pull request merged**